File: pset1/cash/cash.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int	main(void)
{
	float	troco;
	int		centavos;
	int		moedas;
	char	buffer[100];

	moedas = 0;
	do
	{
		printf("Change owed: ");
		fgets(buffer, sizeof(buffer), stdin);
		char *endptr;
		troco = strtof(buffer, &endptr);
		if (endptr != buffer && *endptr == '\n' && troco > 0)
			break;
	} while (endptr == buffer || *endptr != '\n' || troco <= 0);

	centavos = round(troco * 100);
	while (centavos > 0)
	{
		if (centavos >= 25)
			centavos -= 25;
		else if (centavos >= 10)
			centavos -= 10;
		else if (centavos >= 5)
			centavos -= 5;
		else
			centavos -= 1;

		moedas++;
	}

	printf("%d\n", moedas);
	return (0);
}

/* formatado para uso da ide cs50 online

#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    float troco;
    int centavos;
    int moedas = 0;

    do
    {
        troco = get_float("Change owed: ");
    }
    while (troco <= 0);

    centavos = round(troco * 100);

    while (centavos > 0)
    {
        if (centavos >= 25)
            centavos -= 25;
        else if (centavos >= 10)
            centavos -= 10;
        else if (centavos >= 5)
            centavos -= 5;
        else
            centavos -= 1;
        moedas++;
    }

    printf("%d\n", moedas);

    return (0);
}
*/

File: pset6/cash/cash.py
def main():

    while True:
        user_input = input("Change owed: ")
        if user_input.strip() == "":
            continue

        try:
            change = float(user_input)
            if change > 0:
                break
        except ValueError:
            continue

    cents = round(change * 100)
    coins = 0

    while cents > 0:
        if cents >= 25:
            cents -= 25
        elif cents >= 10:
            cents -= 10
        elif cents >= 5:
            cents -=5
        else:
            cents -= 1
        coins += 1

    print(coins)

main()


# formatado para usar a ide cs50
#  import cs50

#  def main():

    #  while True:
        #  change = cs50.get_float("Change owed: ")
        #  if change > 0:
                #  break

    #  cents = round(change * 100)
    #  coins = 0

    #  while cents > 0:
        #  if cents >= 25:
            #  cents -= 25
        #  elif cents >= 10:
            #  cents -= 10
        #  elif cents >= 5:
            #  cents -=5
        #  else:
            #  cents -= 1
        #  coins += 1

    #  print(coins)

#  main()

